.github/workflows/artifacts-download-trigger-build.yml

```diff
@@ -0,0 +1,198 @@
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Download dist
+        uses: actions/download-artifact@v2
+        with:
+          name: build
+          path: artifacts-download-trigger/dist
+      - name: Invoke 'failing' workflow and wait for result using this action
+        id: failing-workflow
+        uses: ./artifacts-download-trigger
+        with:
+          workflow: artifacts-download-trigger-failing.yml
+          gh_token: ${{ secrets.BOT_PAT_TOKEN }}
+          wait-for-completion-interval: 10s
+          wait-for-completion-timeout: 5m
+        continue-on-error: true
+      - run: echo "worflow=${{ toJSON(steps.failing-workflow) }}"
+      - uses: nick-invision/assert-action@v1
+        with:
+          expected: failure
+          actual: ${{ steps.failing-workflow.outputs.workflow-conclusion }}
+      - uses: nick-invision/assert-action@v1
+        with:
+          expected: failure
+          actual: ${{ steps.failing-workflow.outcome }}
+
+  timeout-test:
+    needs: [build]
+    runs-on: ubuntu-latest
+    name: "timeout-test [trigger+wait|by workflow filename|shoud report timed_out]"
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Download dist
+        uses: actions/download-artifact@v2
+        with:
+          name: build
+          path: artifacts-download-trigger/dist
+      - name: Invoke 'timeout' workflow and wait for result using this action
+        id: timeout-workflow
+        uses: ./artifacts-download-trigger
+        with:
+          workflow: artifacts-download-trigger-timeout.yml
+          gh_token: ${{ secrets.BOT_PAT_TOKEN }}
+          wait-for-completion-interval: 10s
+          wait-for-completion-timeout: 30s
+        continue-on-error: true
+      - uses: nick-invision/assert-action@v1
+        with:
+          expected: timed_out
+          actual: ${{ steps.timeout-workflow.outputs.workflow-conclusion }}
+      - uses: nick-invision/assert-action@v1
+        with:
+          expected: failure
+          actual: ${{ steps.timeout-workflow.outcome }}
+
+    # - name: Invoke external workflow using this action
+    #   uses: ./
+    #   with:
+    #     workflow: Deploy To Kubernetes
+    #     repo: benc-uk/dapr-store
+    #     token: ${{ secrets.PERSONAL_TOKEN }}
+    #     ref: master
+
+# This is not need until we have some releases to do
+#
+#  deploy:
+#    needs: [echo-1-test, echo-2-test, long-running-test, failing-test, timeout-test]
+#    runs-on: ubuntu-latest
+#    steps:
+#      - name: Check out repository
+#        uses: actions/checkout@v2
+#      - name: Download dist
+#        uses: actions/download-artifact@v2
+#        with:
+#          name: build
+#          path: artifacts-download-trigger/dist
+#      - name: Update repo with build
+#        uses: mikeal/publish-to-github-action@master
+#        env:
+#          GITHUB_TOKEN: ${{ secrets.BOT_PAT_TOKEN }}
```

Comment:
I do not think we will ever need this. We can use the action with the `@v1` tag after merging the code in the correct branch.